public/app/plugins/datasource/loki/components/useLokiLabels.test.ts

```diff
@@ -1,30 +1,90 @@
 import { renderHook, act } from '@testing-library/react-hooks';
 import LanguageProvider from 'app/plugins/datasource/loki/language_provider';
-import { useLokiLabels } from './useLokiLabels';
+import { getLokiLabels, useLokiLabels } from './useLokiLabels';
 import { AbsoluteTimeRange } from '@grafana/data';
 import { makeMockLokiDatasource } from '../mocks';
+import { CascaderOption } from '@grafana/ui';
 
-describe('useLokiLabels hook', () => {
-  it('should refresh labels', async () => {
-    const datasource = makeMockLokiDatasource({});
-    const languageProvider = new LanguageProvider(datasource);
-    const logLabelOptionsMock = ['Holy mock!'];
-    const rangeMock: AbsoluteTimeRange = {
-      from: 1560153109000,
-      to: 1560153109000,
-    };
+// Mocks
+const datasource = makeMockLokiDatasource({});
+const languageProvider = new LanguageProvider(datasource);
+
+const logLabelOptionsMock = ['Holy mock!'];
+const logLabelOptionsMock2 = ['Mock the hell?!'];
+const logLabelOptionsMock3 = ['Oh my mock!'];
+
+const rangeMock: AbsoluteTimeRange = {
+  from: 1560153109000,
+  to: 1560153109000,
+};
 
+describe('getLokiLabels hook', () => {
+  it('should refresh labels', async () => {
     languageProvider.logLabelOptions = ['initial'];
 
     languageProvider.refreshLogLabels = () => {
       languageProvider.logLabelOptions = logLabelOptionsMock;
       return Promise.resolve();
     };
 
-    const { result, waitForNextUpdate } = renderHook(() => useLokiLabels(languageProvider, true, rangeMock));
+    const { result, waitForNextUpdate } = renderHook(() => getLokiLabels(languageProvider, true, rangeMock));
     expect(result.current.logLabelOptions).toEqual(['initial']);
     act(() => result.current.refreshLabels());
     await waitForNextUpdate();
     expect(result.current.logLabelOptions).toEqual(logLabelOptionsMock);
   });
 });
+
+describe('useLokiLabels hook', () => {
```

Comment:
Tests from the useLokiLabelsAndSyntax that test the log labels fetching and refreshing